templates/default-layout-wrapper.hamlet

```diff
@@ -19,6 +19,28 @@ $newline never
     \<!--[if lt IE 9]>
     \<script src="http://html5shiv.googlecode.com/svn/trunk/html5.js"></script>
     \<![endif]-->
+    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.js">
+    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/js-cookie/2.0.3/js.cookie.min.js">
+
+    <script>
+      
+      console.log("in ajax prefilter.");
+      var csrfCookieName = "#{TE.decodeUtf8 defaultCsrfCookieName}";
+      var csrfHeaderName = "#{TE.decodeUtf8 $ CI.foldedCase defaultCsrfHeaderName}";
+
+      console.log("cookie name = " + csrfCookieName);
+      console.log("header name = " + csrfHeaderName);
+
+      var csrfToken = Cookies.get(csrfCookieName);
+
+      console.log("token = " + csrfToken);
+      if (csrfToken) {
+      \  var defaultHeaders = {};
+      \  defaultHeaders[csrfHeaderName] = csrfToken;
+      \  $.ajaxSetup({
+      \      headers: defaultHeaders,
+      \  });
```

Comment:
I think this should maybe only add the header if the domain the AJAX request is being sent to is the current domain. Need to look into this.